aws-s3-transfer-manager/src/operation/upload.rs

```diff
@@ -42,26 +43,89 @@ impl Upload {
 
         let stream = input.take_body();
         let ctx = new_context(handle, input);
-        let mut handle = UploadHandle::new(ctx);
 
         // MPU has max of 10K parts which requires us to know the upper bound on the content length (today anyway)
         let content_length = stream
             .size_hint()
             .upper()
             .ok_or_else(crate::io::error::Error::upper_bound_size_hint_required)?;
 
-        if content_length < min_mpu_threshold {
-            // TODO - adapt body to ByteStream and send request using `PutObject` for non mpu upload
-            // tracing::trace!("upload request content size hint ({content_length}) less than min part size threshold ({min_mpu_threshold}); sending as single PutObject request");
-            try_start_mpu_upload(&mut handle, stream, content_length).await?
+        let handle = if content_length < min_mpu_threshold {
+            tracing::trace!("upload request content size hint ({content_length}) less than min part size threshold ({min_mpu_threshold}); sending as single PutObject request");
+            try_start_put_object(ctx, stream, content_length).await?
         } else {
-            try_start_mpu_upload(&mut handle, stream, content_length).await?
-        }
+            try_start_mpu_upload(ctx, stream, content_length).await?
+        };
 
         Ok(handle)
     }
 }
 
+async fn try_start_put_object(
+    ctx: UploadContext,
+    stream: InputStream,
+    content_length: u64,
+) -> Result<UploadHandle, crate::error::Error> {
+    let byte_stream = stream.into_byte_stream().await?;
+    let content_length: i64 = content_length.try_into().map_err(|_| {
+        error::invalid_input(format!("content_length:{} is invalid.", content_length))
+    })?;
+
+    Ok(UploadHandle::new_put_object(
+        ctx.clone(),
+        tokio::spawn(put_object(ctx.clone(), byte_stream, content_length)),
+    ))
+}
+
+async fn put_object(
+    ctx: UploadContext,
+    body: ByteStream,
+    content_length: i64,
+) -> Result<UploadOutput, error::Error> {
+    // FIXME - This affects performance in cases with a lot of small files workloads. We need a way to schedule
+    // more work for a lot of small files.
+    let _permit = ctx.handle.scheduler.acquire_permit().await?;
+    let resp = ctx
+        .client()
+        .put_object()
+        .set_acl(ctx.request.acl.clone())
+        .body(body)
+        .set_bucket(ctx.request.bucket.clone())
```

Comment:
fine for now but to keep this and start MPU in sync we may want some kind of macro that converts all the common fields between the two and sets them on the builder.